README.md
# limesurvey-parser
Utilitário para converter JSON do Limesurvey (API Remote Control 2) em JavaScript Object


CHANGELOG.md
Changelog
=========

v0.0.1
------

- First commit ;)

lib/limesurvey-parser.js
/*!
 * @description Recursive object extending
 * @author Viacheslav Lotsmanov <lotsmanov89@gmail.com>
 * @license MIT
 *
 * The MIT License (MIT)
 *
 * Copyright (c) 2013-2015 Viacheslav Lotsmanov
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of
 * this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to
 * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
 * the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

'use strict';

// import { responseDecodedExample, responseBase64Example, questionExample } from './limesurveyExampleData'

const responseBody = require('./samples/answer').responseBase64()
const questionExample = require('./samples/question').questionList()

const buffer = Buffer.from(responseBody.result, 'base64')

const convertBase64ToObject = (encoded) => {
    const decoded = Buffer.from(encoded, 'base64').toString('utf8')
    return JSON.parse(decoded)
}

const getResponses = () => (convertBase64ToObject(responseBody.result)).responses
const getQuestions = () => questionExample.result

const listQuestions = getQuestions()
const listResponses = getResponses()
// console.log(listResponses)


const array = [1,2,3,4,5]
// listQuestions.forEach(item => console.log(item))
// console.log(listQuestions)

let parsedAnswers = []

listResponses.forEach((item, idx) => {
    const response = (Object.values(item))[0]
    const answerFilter = /(E\d)/g
    const limesurveyResponse = Object.entries(response)
    const filteredAnswers = limesurveyResponse
        .filter(val => val[0].match(answerFilter))
        .map(array2d => ({
            field: array2d[0],
            type: null,
            title: null,
            value: array2d[1],
        }))
    const answerHeader = {
        startedAt: response.startdate,
        submittedAt: response.submitdate,
        completed: Boolean(response.submitdate),
        user: null,
        organization: null,
    }

    const result = (answerHeader, filteredAnswers) => {
        let answer = {}
        answer.answers = filteredAnswers
        return Object.assign(answerHeader, answer)
    }

    parsedAnswers.push(result(answerHeader, filteredAnswers))
})

// Aqui vai o teste
//console.log(JSON.stringify(parsedAnswers[0], null, 4))
//console.log(JSON.stringify(listQuestions, null, 4))

//const x = listQuestions[0].title;
//const y = parsedAnswers[100].answers[0].field;

/*listQuestions.forEach((q) => {
    console.log(q.title);
});
*/
parsedAnswers.forEach((a) => {
    if (a.answers) {
        a.answers.forEach((b) => {

            var e = '';
            var sq1 = '';
            var sq2 = '';
            
            var fieldParts = b.field.split('[');
            if (fieldParts.length == 1) {
                e = fieldParts[0];
                console.log(e);
            }
            else if (fieldParts.length == 2) {
                var subfieldParts = fieldParts[1].split('_');
                e = fieldParts[0];
                if (subfieldParts.length == 1) {
                    sq1 = subfieldParts[0].substring(0, subfieldParts[0].length - 1);
                }
                else if (subfieldParts.length == 2) {
                    sq1 = subfieldParts[0];
                    sq2 = subfieldParts[1].substring(0, subfieldParts[1].length - 1);
                }

                console.log(e + ' ---> ' + sq1 + (sq2 ? ' ---> ' + sq2 : ''));
            }
        });
    }
});

//console.log('question: ' + x)
//console.log('answer  : ' + y)



// { id: '147',
//     submitdate: '2017-07-13 16:50:48',
//     lastpage: '1',
//     startlanguage: 'pt-BR',
//     token: 'ucBxxW2J3Kb52TY',
//     startdate: '2017-07-13 16:33:54',
//     datestamp: '2017-07-13 16:50:48',
//     'E0[SQ001_SQ001]': 'HP CONFECCOES HUMBERTO PASCUINI LTDA',
//     'E0[SQ002_SQ001]': 'http://confeccoeshp.com.br/',
//     'E0[SQ003_SQ001]': '220',
//     'E0[SQ004_SQ001]': 'Av. Washington Luiz, 375 e 379, Espírito Santo do Pinhal-SP. CEP: 13.990-000',
//     'E1[SQ001_SQ001]': 'Romildo',
//     'E1[SQ002_SQ001]': 'Barbosa',
//     'E1[SQ003_SQ001]': 'Gerente TI',
//     'E1[SQ004_SQ001]': 'confeccoeshp@confeccoeshp.com.br',
//     'E1[SQ005_SQ001]': '19 36519756',
//     'E2[SQ001]': 'Sim',
//     'E2[SQ002]': 'Não',
//     'E2[SQ003]': 'Não',
//     'E2[SQ004]': 'Não',
//     'E2[SQ005]': 'Não',
//     'E2[other]': '',
//     E3: '1 a 50',
//     'E3[other]': '',
//     'E04[SQ001]': 'Não',
//     'E04[SQ002]': 'Não',
//     'E04[SQ003]': 'Não',
//     'E04[SQ004]': 'Não',
//     'E04[SQ005]': 'Sim',
//     'E04[other]': '',
//     'E05[SQ001]': 'Não',
//     'E05[SQ002]': 'Sim',
//     'E05[SQ003]': 'Sim',
//     'E05[SQ004]': 'Não',
//     'E05[other]': '',
//     E06: 'Não temos nenhum projeto de mudanças',
//     'E06[other]': '',
//     E99: 'Não qualifica',
//     'E99[comment]': 'Empresa familiar, estão satisfeitos com o atual servidor, trabalham com apenas 40 contas, não tem em vista um projeto de mudança.\r\n' }






lib/limesurvey-parser-2.js
const result = require('./samples/limesurvey-json')

var output = []

result.forEach((r) => {

    var obj = {
        "startedAt": r["startdate. Data de início"],
        "submittedAt": r["submitdate. Data de envio"],
        "completed": true,
        "user": null,
        "organization": "BRA:" + r["firstname. Nome"],
        "token": r["token. Código de acesso"],
        "answers": []
    }

    var questions = Object.keys(r).filter(val => val.match(/E\d{1,2}[[.]\s?/))

    questions.forEach((q) => {
        var field = q.substring(0, q.indexOf('.'))
        var type = "short_text"
        
        var title = ""
        if (field.indexOf('SQ') >= 0 || field.indexOf('other') >= 0 || field.indexOf('comment') >= 0)
            title = q.substring(q.indexOf(' [') + 2, q.lastIndexOf(']')).replace(/[\]\[]/g, '')
        else
            title = q.substring(q.indexOf('.') + 2)

        var group = ""
        if (field.indexOf('SQ') >= 0 || field.indexOf('other') >= 0 || field.indexOf('comment') >= 0)
            group = q.substring(q.indexOf('.') + 2, q.indexOf(' ['))
        else
            group = q.substring(q.indexOf('.') + 2)

        var value = r[q];

        obj.answers.push({
                "field": field,
                "type": type,
                "title": title,
                "value": value,
                "group": group
            }
        )
    })

    output.push(obj)
})

console.log(output[100])


lib/samples/question.js

const questionList = () => ({
    "id": 1,
    "result": [{
        "id": {
            "qid": "7",
            "language": "pt-BR"
        },
        "qid": "7",
        "parent_qid": "0",
        "sid": "321796",
        "type": ";",
        "title": "E1",
        "question": "Dados da pessoa de contato",
        "preg": "",
        "help": "Deve ser a pessoa respons\u00e1vel pela infraestrutura de TI da empresa",
        "other": "N",
        "mandatory": "Y",
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "8",
            "language": "pt-BR"
        },
        "qid": "8",
        "parent_qid": "7",
        "sid": "321796",
        "type": ";",
        "title": "SQ001",
        "question": "Nome",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "9",
            "language": "pt-BR"
        },
        "qid": "9",
        "parent_qid": "7",
        "sid": "321796",
        "type": ";",
        "title": "SQ001",
        "question": "",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "1",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "10",
            "language": "pt-BR"
        },
        "qid": "10",
        "parent_qid": "7",
        "sid": "321796",
        "type": ";",
        "title": "SQ002",
        "question": "Sobrenome",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "2",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "11",
            "language": "pt-BR"
        },
        "qid": "11",
        "parent_qid": "7",
        "sid": "321796",
        "type": ";",
        "title": "SQ003",
        "question": "Cargo",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "3",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "12",
            "language": "pt-BR"
        },
        "qid": "12",
        "parent_qid": "7",
        "sid": "321796",
        "type": ";",
        "title": "SQ004",
        "question": "E-mail",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "4",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "13",
            "language": "pt-BR"
        },
        "qid": "13",
        "parent_qid": "7",
        "sid": "321796",
        "type": ";",
        "title": "SQ005",
        "question": "Telefone",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "5",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "14",
            "language": "pt-BR"
        },
        "qid": "14",
        "parent_qid": "0",
        "sid": "321796",
        "type": "M",
        "title": "E2",
        "question": "Gerenciador de e-mail",
        "preg": "",
        "help": "",
        "other": "Y",
        "mandatory": "Y",
        "question_order": "2",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "15",
            "language": "pt-BR"
        },
        "qid": "15",
        "parent_qid": "14",
        "sid": "321796",
        "type": "M",
        "title": "SQ001",
        "question": "Locaweb\/UOLhost",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "16",
            "language": "pt-BR"
        },
        "qid": "16",
        "parent_qid": "14",
        "sid": "321796",
        "type": "M",
        "title": "SQ002",
        "question": "Zimbra",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "2",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "17",
            "language": "pt-BR"
        },
        "qid": "17",
        "parent_qid": "14",
        "sid": "321796",
        "type": "M",
        "title": "SQ003",
        "question": "Kinghost",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "3",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "18",
            "language": "pt-BR"
        },
        "qid": "18",
        "parent_qid": "14",
        "sid": "321796",
        "type": "M",
        "title": "SQ004",
        "question": "Hostgator",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "4",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "19",
            "language": "pt-BR"
        },
        "qid": "19",
        "parent_qid": "14",
        "sid": "321796",
        "type": "M",
        "title": "SQ005",
        "question": "Mandic",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "5",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "20",
            "language": "pt-BR"
        },
        "qid": "20",
        "parent_qid": "0",
        "sid": "321796",
        "type": "L",
        "title": "E3",
        "question": "Quantas contas de e-mail tem aproximadamente?",
        "preg": "",
        "help": "",
        "other": "Y",
        "mandatory": "Y",
        "question_order": "3",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "21",
            "language": "pt-BR"
        },
        "qid": "21",
        "parent_qid": "0",
        "sid": "321796",
        "type": "M",
        "title": "E04",
        "question": "Que aspectos seria interessante melhorar na infra atual?",
        "preg": "",
        "help": "",
        "other": "Y",
        "mandatory": "Y",
        "question_order": "4",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "22",
            "language": "pt-BR"
        },
        "qid": "22",
        "parent_qid": "21",
        "sid": "321796",
        "type": "M",
        "title": "SQ001",
        "question": "Recebemos muito SPAM",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "23",
            "language": "pt-BR"
        },
        "qid": "23",
        "parent_qid": "21",
        "sid": "321796",
        "type": "M",
        "title": "SQ002",
        "question": "Falta de seguran\u00e7a (v\u00edrus, invas\u00f5es, etc)",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "2",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "24",
            "language": "pt-BR"
        },
        "qid": "24",
        "parent_qid": "21",
        "sid": "321796",
        "type": "M",
        "title": "SQ003",
        "question": "Lentid\u00e3o",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "3",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "25",
            "language": "pt-BR"
        },
        "qid": "25",
        "parent_qid": "21",
        "sid": "321796",
        "type": "M",
        "title": "SQ004",
        "question": "Falta de ferramentas de colabora\u00e7\u00e3o (videoconfer\u00eancias, documentos, treinamentos, etc)",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "4",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "26",
            "language": "pt-BR"
        },
        "qid": "26",
        "parent_qid": "21",
        "sid": "321796",
        "type": "M",
        "title": "SQ005",
        "question": "N\u00e3o temos problemas. Est\u00e1 tudo bem.",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "5",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "27",
            "language": "pt-BR"
        },
        "qid": "27",
        "parent_qid": "0",
        "sid": "321796",
        "type": "M",
        "title": "E05",
        "question": "Como controlam o SPAM?",
        "preg": "",
        "help": "",
        "other": "Y",
        "mandatory": "Y",
        "question_order": "5",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "28",
            "language": "pt-BR"
        },
        "qid": "28",
        "parent_qid": "27",
        "sid": "321796",
        "type": "M",
        "title": "SQ001",
        "question": "Contratamos uma ferramenta antiSpam",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "29",
            "language": "pt-BR"
        },
        "qid": "29",
        "parent_qid": "27",
        "sid": "321796",
        "type": "M",
        "title": "SQ002",
        "question": "Operamos com o antiSpam do fornecedor de e-mail",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "2",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "30",
            "language": "pt-BR"
        },
        "qid": "30",
        "parent_qid": "27",
        "sid": "321796",
        "type": "M",
        "title": "SQ003",
        "question": "Controlamos manualmente",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "3",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "31",
            "language": "pt-BR"
        },
        "qid": "31",
        "parent_qid": "27",
        "sid": "321796",
        "type": "M",
        "title": "SQ004",
        "question": "N\u00e3o controlamos, pois, n\u00e3o recebemos",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "4",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "32",
            "language": "pt-BR"
        },
        "qid": "32",
        "parent_qid": "0",
        "sid": "321796",
        "type": "L",
        "title": "E06",
        "question": "H\u00e1 algum projeto de melhoria de infraestrutura?",
        "preg": "",
        "help": "",
        "other": "Y",
        "mandatory": "Y",
        "question_order": "6",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "33",
            "language": "pt-BR"
        },
        "qid": "33",
        "parent_qid": "0",
        "sid": "321796",
        "type": "O",
        "title": "E99",
        "question": "Qualifica\u00e7\u00e3o do Lead",
        "preg": "",
        "help": "",
        "other": "N",
        "mandatory": "Y",
        "question_order": "7",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": "",
        "gid": "3"
    }, {
        "id": {
            "qid": "34",
            "language": "pt-BR"
        },
        "qid": "34",
        "parent_qid": "0",
        "sid": "321796",
        "type": ";",
        "title": "E0",
        "question": "Sobre a empresa",
        "preg": "",
        "help": "",
        "other": "N",
        "mandatory": "Y",
        "question_order": "0",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "35",
            "language": "pt-BR"
        },
        "qid": "35",
        "parent_qid": "34",
        "sid": "321796",
        "type": ";",
        "title": "SQ001",
        "question": "Raz\u00e3o social",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "36",
            "language": "pt-BR"
        },
        "qid": "36",
        "parent_qid": "34",
        "sid": "321796",
        "type": ";",
        "title": "SQ001",
        "question": "",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "1",
        "language": "pt-BR",
        "scale_id": "1",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "37",
            "language": "pt-BR"
        },
        "qid": "37",
        "parent_qid": "34",
        "sid": "321796",
        "type": ";",
        "title": "SQ002",
        "question": "Site",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "2",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "38",
            "language": "pt-BR"
        },
        "qid": "38",
        "parent_qid": "34",
        "sid": "321796",
        "type": ";",
        "title": "SQ003",
        "question": "N\u00famero de funcion\u00e1rios",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "3",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }, {
        "id": {
            "qid": "39",
            "language": "pt-BR"
        },
        "qid": "39",
        "parent_qid": "34",
        "sid": "321796",
        "type": ";",
        "title": "SQ004",
        "question": "Endere\u00e7o completo",
        "preg": null,
        "help": null,
        "other": "N",
        "mandatory": null,
        "question_order": "4",
        "language": "pt-BR",
        "scale_id": "0",
        "same_default": "0",
        "relevance": "1",
        "modulename": null,
        "gid": "3"
    }],
    "error": null
})


module.exports = { questionList }